thunder/__init__.py

```diff
@@ -369,7 +369,7 @@ def _alias_tensor_of_args_kwargs_dict(*args, **kwargs) -> dict[int, list[int]]:
         data_ptr_to_tensor_group_index = {}
         tensor_group_index_to_tensor_indices = defaultdict(list)
         for idx, t in enumerate(flat_args):
-            if pytorch.is_tensor(t) and t.layout == pytorch.strided:
+            if type(t) in {pytorch.Tensor, pytorch.nn.Parameter} and t.layout == pytorch.strided:
```

Comment:
Could you please submit this fix with a test in a separate pull request?

Reply:
That pull request would need a subclass then I'm not quite convinced by the option

Reply:
How do subclasses appear here? Do all subclasses have the actual torch.Tensor type?